src/figma-code/steps/headings.js

```diff
@@ -109,36 +109,186 @@ export const listener = (msg) => {
     }
   }
 
-  const { mainPageId, name } = page;
+  // listen flag: see figma.on('selectionchange')
```

Comment:
I actually had already added the cursor focus hack in the code above at some point during the v3 work (expand lines 94-108 which I didn't update here), but it didn't really come in as handy until now